### DISCUSSION TITLE
vector search return empty result【向量检索的时候返回空结果】

### DISCUSSION THREAD

**Discussion by 250465043** [milvus-io/milvus#32667] [Q&A and General discussion] [answered]
版本：milvus 2.3
环境：阿里云云主机
数据量：10万条
检索方式：余弦相似度

使用标量过滤，向量检索，某条数据始终无法检索出来，我通过golang模拟请求，以及在attu上做vector query都检测不到该条数据。

我在云主机上做了另外一个集合，只有这一条数据，是可以检索出来的

我在本地环境下（开发用的pc）上搭建的同样版本的milvus，导入同样的数据，完全一样的检索请求，也是可以检索出来的。

请问原因是什么？感谢！

**Comment by yhmo** [comment, marked as answer]
如果建了IVF/HNSW索引的话，是有可能搜不出某些符合条件的值。你换成 FLAT索引肯定就能搜出来。

**Reply by 250465043** [reply]
看上确实是这个原因，感谢~
